apps/files_external/lib/Lib/Storage/SMB.php

```
@@ -296,112 +329,141 @@ public function rmdir($path) {
+				$result = false;
 			}
-			return false;
 		} catch (ConnectException $e) {
-			throw new StorageNotAvailableException($e->getMessage(), $e->getCode(), $e);
+			throw $this->leave(__FUNCTION__, new StorageNotAvailableException(
+				$e->getMessage(), $e->getCode(), $e));
 		}
+		return $this->leave(__FUNCTION__, $result);
 	}
 
 	public function opendir($path) {
+		$this->log('enter: '.__FUNCTION__."($path)");
+		$result = false;
 		try {
 			$files = $this->getFolderContents($path);
+			$names = array_map(function ($info) {
+				/** @var \Icewind\SMB\IFileInfo $info */
+				return $info->getName();
+			}, $files);
+			$result = IteratorDirectory::wrap($names);
 		} catch (NotFoundException $e) {
-			return false;
+			$this->swallow(__FUNCTION__, $e);
 		} catch (ForbiddenException $e) {
-			return false;
+			$this->swallow(__FUNCTION__, $e);
 		}
-		$names = array_map(function ($info) {
-			/** @var \Icewind\SMB\IFileInfo $info */
-			return $info->getName();
-		}, $files);
-		return IteratorDirectory::wrap($names);
+		return $this->leave(__FUNCTION__, $result);
 	}
 
 	public function filetype($path) {
+		$this->log('enter: '.__FUNCTION__."($path)");
+		$result = false;
 		try {
-			return $this->getFileInfo($path)->isDirectory() ? 'dir' : 'file';
+			$result = $this->getFileInfo($path)->isDirectory() ? 'dir' : 'file';
 		} catch (NotFoundException $e) {
-			return false;
+			$this->swallow(__FUNCTION__, $e);
 		} catch (ForbiddenException $e) {
-			return false;
+			$this->swallow(__FUNCTION__, $e);
 		}
+		return $this->leave(__FUNCTION__, $result);
 	}
 
 	public function mkdir($path) {
+		$this->log('enter: '.__FUNCTION__."($path)");
+		$result = false;
 		$path = $this->buildPath($path);
 		try {
 			$this->share->mkdir($path);
-			return true;
+			$result = true;
 		} catch (ConnectException $e) {
-			throw new StorageNotAvailableException($e->getMessage(), $e->getCode(), $e);
+			throw $this->leave(__FUNCTION__, new StorageNotAvailableException(
```

no, because these exceptions are sometimes swallowed, eg. by the availability wrapper: https://github.com/owncloud/core/blob/master/lib/private/Files/Storage/Wrapper/Availability.php#L53.

leave() detects an exception and also logs a stacktrace which is the most interesting thing when debugging this.